roles/confluent.common/defaults/main.yml

```diff
@@ -46,3 +46,12 @@ confluent_cli_download_enabled: "{{rbac_enabled}}"
 
 ### Full path on hosts to install the Confluent CLI
 confluent_cli_path: /usr/local/bin/confluent
+
+### Base URL for Confluent's Archive Repositories
+confluent_common_repository_archive_baseurl: "{{confluent_common_repository_baseurl}}/archive/{{confluent_repo_version}}"
+
+### Name of the file to download from the Archive base URL
+confluent_archive_file_name: "confluent-{{confluent_archive_version}}-{{confluent_archive_scala_version}}.tar.gz"
+
+### A path reference to a local archive file or URL. By default this is the URL from Confluent's repositories. In an ansible-pull deployment this could be set to a local file such as "~/.ansible/pull/{{inventory_hostname}}/{{confluent_archive_file_name}}".
+confluent_archive_file_path: "{{confluent_common_repository_archive_baseurl}}/{{confluent_archive_file_name}}"
```

Comment:
unused vars, please remove

Reply:
`confluent_archive_file_name` is used by `confluent_archive_file_path`
`confluent_archive_file_path` is used in roles/confluent.common/tasks/main.yml

Reply:
ah yes, my point is why have three variables? it could be

```
### A path reference to a local archive file or URL. By default this is the URL from Confluent's repositories. In an ansible-pull deployment this could be set to a local file such as "~/.ansible/pull/{{inventory_hostname}}/confluent-5.5.1-2.12.tar.gz".
confluent_archive_file_path: "{{confluent_common_repository_baseurl}}/archive/{{confluent_repo_version}}/confluent-{{confluent_package_version}}-{{confluent_archive_scala_version}}.tar.gz"
```

Ultimately if someone wants to customize the url for the tar, then they should just update the whole url, allowing them to customize each seems like room for user error... 

Maybe I'm not thinking of the ansible-pull use case. You could just update the comment to 

WDYT?

Reply:
Re: combine variables - that's fine.

ansible-pull gets the tarball from the git repo. In fact it really is just a `git clone` for a specific branch and `ansible-playbook`. There may be better ways of storing it than in the git repo, like internal http, ftp or nfs servers.

ansible-pull really shines when ssh access isn't readily available to the target node.

Reply:
I'm still not seeing it. Lets say we have: 

- 1 ansible control nodes which runs the all.yml playbook
- 3 target nodes that you want to install zookeeper and kafka on

If the control node can do ansible-pull then it must be able to do `git clone` / `git checkout`

Now theres the question of how to get the tarball unarchived on the target nodes. And i see four options:

1. target nodes have internet access and pull directly from confluent's URL
2 target nodes do not have outside internet access, but do have access to a webserver, so you host the tar in a webserver and update the `confluent_archive_file_path` var to that URL
3. target nodes do not have outside internet access, you do not want to stand up a webserver, so you download the tarball onto the control node and set `confluent_archive_file_path` var to the path, and use `remote_src: false` in the unarchive module (we should probably offer this btw)
4. you distribute the tarball somehow to all the target nodes and set `confluent_archive_file_path` var to the path, and use `remote_src: true` in the unarchive module

It seems you are looking to do number 4 in this PR and claiming that ansible-pull helps, and im not seeing how, as well as why you would want to distribute the tarball instead of go with number 3

and when you say:

> ansible-pull really shines when ssh access isn't readily available to the target node.

How does that work? Are you saying there is no control node? 
You only have 3 target nodes and you run the playbooks from themselves? like using `--limit` to limit the run to itself

Sorry for all the comments btw. I think this PR is really close to being ready!

Reply:
No worries on the comments, I think it's really close too. In fact, I don't see any more unresolved comments...

`ansible-pull` isn't run on the control node, it's run directly on the target nodes. And yes, as you have surmised we were leaning toward number 4. However, I am off that engagement now and simply want to offer different methods for administrators in the future.

Again, you are correct, there is no control node. Everything is put into a version control system that is used to source the playbooks to the target nodes. This generally works because there is normally some sort of VCS in an enterprise system where we can version control infrastructure configurations and reliably build/rebuild deployments.

Hope that helps!